database/data-generator/generate-pub-discounts-couch.js

```diff
@@ -19,27 +16,22 @@ const {writeData} = require('../writeToCSV.js');
 const generatePublisherDiscounts = () => {
 
   //create 14 day range promotional discount
-  let days = faker.random.number({min: 1, max: 183}); //gives a random number of days within a 6 month period
+  let days = faker.random.number({min: 1, max: 90}); //gives a random number of days within a 3 month period
   let start = moment().add(days, 'days');
   let end = moment(start).add(14, 'days');
 
-  //format date to month, day, year
-  start = moment(start).format('YYYYMMDD');
-  end = moment(end).format('YYYYMMDD');
+  start = moment(start).format('YYYY-MM-DD');
+  end = moment(end).format('YYYY-MM-DD');
 
   //a row of data
   let record = {
     discount:  faker.random.number({min: 5, max: 95}),
-    publisher_id:  faker.random.number({min: 1, max: 5e6}), //1-5M publisherId's,
     start: start,
-    end: end,
-    type: "publisher_discount"
+    end: end
```

Comment:
Since the general discount and publisher discount code are really similar, it might be better to have generateDiscount() and pass a flag to switch between publisher and general.

Reply:
Thanks for the great suggestion! I I was using the functions that I created for postgres so definitely lazy coding on my part.